lib/wallaroo/core/source/connector_source/connector_source_coordinator.pony

```diff
@@ -435,29 +423,49 @@ actor ConnectorSourceListener[In: Any val] is SourceListener
     _start_sources()
 
   be report_initialized() =>
-    @printf[I32]("ConnectorSourceListener for: %s reporting initialized.\n".cstring(), _pipeline_name.cstring())
+    @printf[I32]("ConnectorSourceCoordinator for: %s reporting initialized.\n"
+      .cstring(), _pipeline_name.cstring())
     _is_joining = false
     try
       (_initializer as LocalTopologyInitializer).report_initialized(this)
     else
       Fail()
     end
 
+  //////////////
+  // BARRIER
+  //////////////
+  be initiate_barrier(token: BarrierToken) =>
```

Comment:
There's one more thing we need to do here:
checkpoint the state of the global stream registry via the local registry (`_stream_registry`).
Once this state is part of the recovery data, we should be able to tolerate leader worker failures as well

Reply:
I don't think there's currently a `serialise/deserialise` implementation for `GlobalConnectorStreamRegistry`, so we'll have to add that in the process.
This can be a separate issue/PR, though

Reply:
Yeah, this PR is just the initial step of bringing SourceCoordinator into the protocol.  The Connector specific part is next.